packages/graphql-language-service-server/src/languageUtils.ts

```diff
@@ -0,0 +1,107 @@
+import { Position, Range } from 'graphql-language-service-utils';
+
+type StrRange = {
+  start: number;
+  end: number;
+  line: number;
+};
+
+type Location = {
+  line: number;
+  column: number;
+  character: number;
+};
+
+export type GraphQLSource = {
+  template: string;
+  range: Range;
+};
+
+export function getLocator(source: string) {
+  const offsetLine = 0;
+  const offsetColumn = 0;
+
+  const originalLines = source.split('\n');
+
+  let start = 0;
+  const lineRanges = originalLines.map((line, i) => {
+    const end = start + line.length + 1;
+    const range: StrRange = { start, end, line: i };
+
+    start = end;
+    return range;
+  });
+
+  let i = 0;
+
+  function rangeContains(range: StrRange, index: number) {
+    return range.start <= index && index < range.end;
+  }
+
+  function getLocation(range: StrRange, index: number): Location {
+    return {
+      line: offsetLine + range.line,
+      column: offsetColumn + index - range.start,
```

Comment:
from what i understand, you could change `offsetColumn` here to 0 and remove line 22 and it would not change anything? is this or `offsetLine` meant for something? or is line 21/22 there for configurability/visibility? just curious more than anything

Reply:
This is actually copied/vendored from https://github.com/Rich-Harris/locate-character so I don't really dare change anything unless absolutely necessary :grin: